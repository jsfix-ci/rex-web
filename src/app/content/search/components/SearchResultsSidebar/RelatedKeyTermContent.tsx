import React from 'react';
import { KeyTermHit } from '../../types';
import * as Styled from './styled';

// tslint:disable-next-line: variable-name
const RelatedKeyTermContent = ({ keyTermHit }: { keyTermHit: KeyTermHit }) => <Styled.KeyTermContainer tabIndex={-1}>
  <Styled.KeyTerm
    dangerouslySetInnerHTML={{ __html: keyTermHit.highlight.term }} />
  <div dangerouslySetInnerHTML={{ __html: keyTermHit.highlight.visibleContent[0] }} />
</Styled.KeyTermContainer>;

export default RelatedKeyTermContent;
