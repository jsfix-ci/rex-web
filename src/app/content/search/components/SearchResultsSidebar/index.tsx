import React, { Component } from 'react';
import { connect } from 'react-redux';
import { AppState, Dispatch } from '../../../../types';
import * as select from '../../../selectors';
import { Book, Page } from '../../../types';
import { clearSearch, closeSearchResultsMobile } from '../../actions';
import * as selectSearch from '../../selectors';
import { SearchResultContainer } from '../../types';
import { SearchResultsBarWrapper } from './SearchResultsBarWrapper';

interface SearchResultsSidebarProps {
  currentPage: Page | undefined;
  book?: Book;
  query: string | null;
  totalHits: number | null;
  results: SearchResultContainer[] | null;
  onClose: () => void;
  searchResultsOpen: boolean;
  closeSearchResults: () => void;
}

export class SearchResultsSidebar extends Component<SearchResultsSidebarProps> {
  public render() {
    const {query} = this.props;

    return query && <SearchResultsBarWrapper {...this.props} />;
  }
}

export default connect(
  (state: AppState) => ({
    book: select.book(state),
    currentPage: select.page(state),
    query: selectSearch.query(state),
    results: selectSearch.results(state),
    searchResultsOpen: selectSearch.searchResultsOpen(state),
    totalHits: selectSearch.totalHits(state),
  }),
  (dispatch: Dispatch) => ({
    closeSearchResults: () => {
      dispatch(closeSearchResultsMobile());
    },
    onClose: () => {
      dispatch(clearSearch());
    },
  })
)(SearchResultsSidebar);
