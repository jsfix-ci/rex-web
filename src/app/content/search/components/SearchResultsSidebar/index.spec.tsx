import React from 'react';
import { unmountComponentAtNode } from 'react-dom';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import SearchResultsSidebar from '.';
import createTestStore from '../../../../../test/createTestStore';
import { book as archiveBook, page, pageInChapter, pageInOtherChapter } from '../../../../../test/mocks/archiveLoader';
import { mockCmsBook } from '../../../../../test/mocks/osWebLoader';
import { makeEvent, makeFindByTestId, makeFindOrNullByTestId, renderToDom } from '../../../../../test/reactutils';
import { makeSearchResultHit, makeSearchResults } from '../../../../../test/searchResults';
import MessageProvider from '../../../../MessageProvider';
import { Store } from '../../../../types';
import { receiveBook, receivePage } from '../../../actions';
import { formatBookData } from '../../../utils';
import { clearSearch, closeSearchResultsMobile, receiveSearchResults, requestSearch } from '../../actions';

describe('SearchResultsSidebar', () => {
  let store: Store;
  let dispatch: jest.SpyInstance;

  beforeEach(() => {
    store = createTestStore();
    dispatch = jest.spyOn(store, 'dispatch');
    store.dispatch(receiveBook(formatBookData(archiveBook, mockCmsBook)));
  });

  const render = () => <MessageProvider>
    <Provider store={store}>
      <SearchResultsSidebar />
    </Provider>
  </MessageProvider>;

  it('mounts and unmounts without a dom', () => {
    store.dispatch(requestSearch('cool search'));
    const component = renderer.create(render());
    expect(() => component.unmount()).not.toThrow();
  });

  it('mounts and unmmounts with a dom', () => {
    store.dispatch(requestSearch('cool search'));
    const {root} = renderToDom(render());
    expect(() => unmountComponentAtNode(root)).not.toThrow();
  });

  it('doesn\'t render if there is no search', () => {
    const component = renderer.create(render());
    const findOrNull = makeFindOrNullByTestId(component.root);

    expect(findOrNull('search-results-sidebar')).toBe(null);
  });

  it('shows sidebar with loading state if there is a search', () => {
    store.dispatch(requestSearch('cool search'));
    const component = renderer.create(render());
    const findOrNull = makeFindOrNullByTestId(component.root);

    expect(findOrNull('search-results-sidebar')).not.toBe(null);
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('matches snapshot for no search results', () => {
    store.dispatch(requestSearch('cool search'));
    store.dispatch(receiveSearchResults(makeSearchResults([])));

    const tree = renderer.create(render()).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('matches snapshot with results', () => {
    store.dispatch(receivePage({...pageInChapter, references: []}));
    store.dispatch(requestSearch('cool search'));
    store.dispatch(receiveSearchResults(makeSearchResults([
      makeSearchResultHit({book: archiveBook, page}),
      makeSearchResultHit({book: archiveBook, page: pageInChapter}),
      makeSearchResultHit({book: archiveBook, page: pageInOtherChapter}),
    ])));

    const tree = renderer.create(render()).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('closes search results when one is clicked', () => {
    store.dispatch(requestSearch('cool search'));
    store.dispatch(receiveSearchResults(makeSearchResults([makeSearchResultHit({book: archiveBook, page})])));

    const component = renderer.create(render());
    const findById = makeFindByTestId(component.root);

    expect(dispatch).not.toHaveBeenCalledWith(closeSearchResultsMobile());

    renderer.act(() => {
      findById('search-result').props.onClick(makeEvent());
    });

    expect(dispatch).toHaveBeenCalledWith(closeSearchResultsMobile());
  });

  it('clears search when closed', () => {
    store.dispatch(requestSearch('cool search'));
    store.dispatch(receiveSearchResults(makeSearchResults([makeSearchResultHit({book: archiveBook, page})])));

    const component = renderer.create(render());
    const findById = makeFindByTestId(component.root);

    expect(dispatch).not.toHaveBeenCalledWith(clearSearch());

    renderer.act(() => {
      findById('close-search').props.onClick(makeEvent());
    });

    expect(dispatch).toHaveBeenCalledWith(clearSearch());
  });
});
