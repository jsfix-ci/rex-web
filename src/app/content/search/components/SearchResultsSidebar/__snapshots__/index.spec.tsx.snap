// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResultsSidebar matches snapshot for no search results 1`] = `
.c4 {
  display: inline-block;
  vertical-align: -.125em;
  overflow: hidden;
}

.c3 {
  height: 2.5rem;
  width: 2.5rem;
  padding: 0.4rem;
  color: #818181;
  margin-right: 1.4rem;
}

.c0 {
  top: 12rem;
  margin-top: 0;
  padding: 0;
  position: -webkit-sticky;
  position: sticky;
  width: 43.5rem;
  background-color: #f1f1f1;
  box-shadow: 0.2rem 0 0.2rem 0 rgba(0,0,0,0.1);
  z-index: 2;
  height: calc(100vh - 18rem);
}

.c0 > .c7 {
  margin: 0;
  padding: 0;
}

.c0 > .c7::before {
  display: none;
}

.c5 {
  font-family: Neue Helvetica W01;
  color: #424242;
  font-size: 1.6rem;
  line-height: 2.5rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: end;
  -webkit-justify-content: end;
  -ms-flex-pack: end;
  justify-content: end;
  min-height: 4rem;
}

.c5 strong {
  padding-left: 0.4rem;
}

.c6 {
  max-width: 26.5rem;
  text-align: justify;
  margin-top: 7rem;
  margin-left: auto;
  margin-right: auto;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
}

.c2 {
  border: none;
  padding: 0;
  margin: 0;
  background: transparent;
}

@media screen and (max-width:75em) {
  .c0 {
    width: 100%;
    margin-top: 0;
    top: 15.3rem;
    padding: 0;
  }
}

@media screen and (max-width:75em) {
  .c1 {
    display: none;
  }
}

@media screen and (max-width:75em) {
  .c2 {
    display: none;
  }
}

<div
  aria-label="Search results sidebar"
  className="c0"
  data-testid="search-results-sidebar"
>
  <div>
    <div
      className="c1"
    >
      <button
        className="c2"
        onClick={[Function]}
      >
        <svg
          aria-hidden="true"
          className="c3 c4"
          fill="currentColor"
          focusable="false"
          viewBox="0 0 352 512"
        >
          <path
            d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
            fill="currentColor"
          />
        </svg>
      </button>
    </div>
    <div
      className="c5"
    >
      <div
        className="c6"
      >
        Sorry, no results found for 
         
        <strong>
           ‘
          cool search
          ’
        </strong>
      </div>
    </div>
  </div>
</div>
`;

exports[`SearchResultsSidebar matches snapshot with results 1`] = `
.c4 {
  display: inline-block;
  vertical-align: -.125em;
  overflow: hidden;
}

.c18 {
  height: 1.7rem;
  width: 1.7rem;
}

.c20 {
  height: 1.7rem;
  width: 1.7rem;
}

.c14[open] > summary .c17 {
  display: none;
}

.c14:not([open]) > summary .c19 {
  display: none;
}

.c22 {
  font-family: Neue Helvetica W01;
  color: #424242;
  font-size: 1.4rem;
  line-height: 1.6rem;
  font-weight: normal;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c16 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c3 {
  height: 2.5rem;
  width: 2.5rem;
  padding: 0.4rem;
  color: #424242;
  margin-right: 0.7rem;
  margin-left: 3rem;
}

.c7 {
  height: 2.5rem;
  width: 2.5rem;
  padding: 0.4rem;
  color: #818181;
  margin-right: 1.4rem;
}

.c9 .os-divider {
  width: 0.4rem;
}

.c0 {
  top: 12rem;
  margin-top: 0;
  padding: 0;
  position: -webkit-sticky;
  position: sticky;
  width: 43.5rem;
  background-color: #f1f1f1;
  box-shadow: 0.2rem 0 0.2rem 0 rgba(0,0,0,0.1);
  z-index: 2;
  height: calc(100vh - 18rem);
}

.c0 > .c8 {
  margin: 0;
  padding: 0;
}

.c0 > .c8::before {
  display: none;
}

.c2 {
  font-family: Neue Helvetica W01;
  color: #424242;
  font-size: 1.6rem;
  line-height: 2.5rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: end;
  -webkit-justify-content: end;
  -ms-flex-pack: end;
  justify-content: end;
  min-height: 4rem;
}

.c2 strong {
  padding-left: 0.4rem;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: baseline;
  -webkit-box-align: baseline;
  -ms-flex-align: baseline;
  align-items: baseline;
  background: #fff;
  padding: 1rem 0;
}

.c15 {
  list-style: none;
  cursor: pointer;
  min-height: 3.8rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: #f1f1f1;
  border-top: solid 0.1rem #d5d5d5;
  padding-left: 3rem;
}

.c15::before {
  display: none;
}

.c15::-moz-list-bullet {
  list-style-type: none;
}

.c15::-webkit-details-marker {
  display: none;
}

.c15:focus {
  outline: none;
}

.c15:hover .c21,
.c15:focus .c21 {
  color: #000;
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c12 {
  font-family: Neue Helvetica W01;
  color: #424242;
  font-size: 1.4rem;
  line-height: 1.6rem;
  font-weight: normal;
  width: 100%;
}

.c13 {
  font-family: Neue Helvetica W01;
  color: #424242;
  font-size: 1.4rem;
  line-height: 1.6rem;
  font-weight: normal;
  margin-left: 7rem;
  min-height: 3.7rem;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-right: 1.6rem;
  padding: 1rem 0;
  max-width: 35.3rem;
  display: block;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c13:not(:last-child) {
  border-bottom: solid 0.1rem #f1f1f1;
}

.c13 em {
  font-weight: bold;
}

.c11 {
  min-height: 3.4rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-left: 4.7rem;
  padding-top: 1rem;
  padding-bottom: 0.6rem;
}

.c23 {
  padding: 0;
}

.c10 {
  background: #fff;
}

.c10:not(:last-child) {
  border-bottom: solid 0.2rem #f1f1f1;
}

.c6 {
  border: none;
  padding: 0;
  margin: 0;
  background: transparent;
}

.c5 {
  max-width: calc(34.1rem);
}

@media screen and (max-width:75em) {
  .c0 {
    width: 100%;
    margin-top: 0;
    top: 15.3rem;
    padding: 0;
  }
}

@media screen and (max-width:75em) {
  .c6 {
    display: none;
  }
}

<div
  aria-label="Search results sidebar"
  className="c0"
  data-testid="search-results-sidebar"
>
  <div
    className="c1"
  >
    <div
      className="c2"
    >
      <svg
        aria-hidden="true"
        className="c3 c4"
        fill="currentColor"
        focusable="false"
        viewBox="0 0 512 512"
      >
        <path
          d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
          fill="currentColor"
        />
      </svg>
      <div
        className="c5"
      >
        3
         
        search results for 
         
        <strong>
           ‘
          cool search
          ’
        </strong>
      </div>
    </div>
    <button
      aria-label="Close search sidebar"
      className="c6"
      data-testid="close-search"
      onClick={[Function]}
    >
      <svg
        aria-hidden="true"
        className="c7 c4"
        fill="currentColor"
        focusable="false"
        viewBox="0 0 352 512"
      >
        <path
          d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
          fill="currentColor"
        />
      </svg>
    </button>
  </div>
  <ol
    className="c8 c9"
  >
    <li
      className="c10"
    >
      <div
        className="c11"
      >
        <div
          className="c12"
          dangerouslySetInnerHTML={
            Object {
              "__html": "<span class=\\"os-text\\">Test Page 1</span>",
            }
          }
        />
      </div>
      <a
        className="c13"
        dangerouslySetInnerHTML={
          Object {
            "__html": "cool <em>highlight</em> bruh",
          }
        }
        data-testid="search-result"
        href="books/book-slug-1/pages/test-page-1"
        onClick={[Function]}
      />
    </li>
    <li>
      <details
        className="c14"
        open={true}
      >
        <summary
          className="c15"
        >
          <div
            className="c16"
          >
            <svg
              aria-hidden="true"
              className="c17 c18 c4"
              fill="currentColor"
              focusable="false"
              viewBox="0 0 192 512"
            >
              <path
                d="M0 384.662V127.338c0-17.818 21.543-26.741 34.142-14.142l128.662 128.662c7.81 7.81 7.81 20.474 0 28.284L34.142 398.804C21.543 411.404 0 402.48 0 384.662z"
                fill="currentColor"
              />
            </svg>
            <svg
              aria-hidden="true"
              className="c19 c20 c4"
              fill="currentColor"
              focusable="false"
              viewBox="0 0 320 512"
            >
              <path
                d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
                fill="currentColor"
              />
            </svg>
            <span
              className="c21 c22"
              dangerouslySetInnerHTML={
                Object {
                  "__html": "<span class=\\"os-number\\">10</span><span class=\\"os-divider\\"> </span><span class=\\"os-text\\">Test Chapter 5</span>",
                }
              }
            />
          </div>
        </summary>
        <ol
          className="c23"
        >
          <li
            className="c10"
          >
            <div
              aria-label="Current Page"
              className="c11"
            >
              <div
                className="c12"
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "<span class=\\"os-text\\">Test Page 6</span>",
                  }
                }
              />
            </div>
            <a
              className="c13"
              dangerouslySetInnerHTML={
                Object {
                  "__html": "cool <em>highlight</em> bruh",
                }
              }
              data-testid="search-result"
              href="books/book-slug-1/pages/10-test-page-6"
              onClick={[Function]}
            />
          </li>
        </ol>
      </details>
    </li>
    <li>
      <details
        className="c14"
      >
        <summary
          className="c15"
        >
          <div
            className="c16"
          >
            <svg
              aria-hidden="true"
              className="c17 c18 c4"
              fill="currentColor"
              focusable="false"
              viewBox="0 0 192 512"
            >
              <path
                d="M0 384.662V127.338c0-17.818 21.543-26.741 34.142-14.142l128.662 128.662c7.81 7.81 7.81 20.474 0 28.284L34.142 398.804C21.543 411.404 0 402.48 0 384.662z"
                fill="currentColor"
              />
            </svg>
            <svg
              aria-hidden="true"
              className="c19 c20 c4"
              fill="currentColor"
              focusable="false"
              viewBox="0 0 320 512"
            >
              <path
                d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
                fill="currentColor"
              />
            </svg>
            <span
              className="c21 c22"
              dangerouslySetInnerHTML={
                Object {
                  "__html": "<span class=\\"os-number\\">12</span><span class=\\"os-divider\\"> </span><span class=\\"os-text\\">Test Chapter 6</span>",
                }
              }
            />
          </div>
        </summary>
        <ol
          className="c23"
        >
          <li
            className="c10"
          >
            <div
              className="c11"
            >
              <div
                className="c12"
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "<span class=\\"os-text\\">Test Page 7</span>",
                  }
                }
              />
            </div>
            <a
              className="c13"
              dangerouslySetInnerHTML={
                Object {
                  "__html": "cool <em>highlight</em> bruh",
                }
              }
              data-testid="search-result"
              href="books/book-slug-1/pages/12-test-page-7"
              onClick={[Function]}
            />
          </li>
        </ol>
      </details>
    </li>
  </ol>
</div>
`;

exports[`SearchResultsSidebar shows sidebar with loading state if there is a search 1`] = `
.c6 {
  enable-background: new 0 0 57.6 39.1;
  -webkit-animation: iuEIAH 0.3s forwards 0.3s;
  animation: iuEIAH 0.3s forwards 0.3s;
  display: block;
  height: 100%;
  margin: 0 auto;
  opacity: 0;
  position: relative;
  -webkit-transform: scale(0);
  -ms-transform: scale(0);
  transform: scale(0);
  -webkit-transform-origin: center;
  -ms-transform-origin: center;
  transform-origin: center;
  width: 5rem;
}

.c6 * {
  -webkit-animation: fadeIn 0.3s forwards;
  animation: fadeIn 0.3s forwards;
}

.c6 .os-green,
.c6 .os-orange,
.c6 .os-gray,
.c6 .os-yellow,
.c6 .os-blue {
  -webkit-animation-delay: 0.5s;
  animation-delay: 0.5s;
}

.c6 .os-green {
  -webkit-animation: hkCfKg 0.6s cubic-bezier(0.81,0.41,0.13,0.74) 0.5s infinite alternate;
  animation: hkCfKg 0.6s cubic-bezier(0.81,0.41,0.13,0.74) 0.5s infinite alternate;
  -webkit-transform-origin: center left;
  -ms-transform-origin: center left;
  transform-origin: center left;
}

.c6 .os-orange {
  -webkit-animation: kSacDd 0.6s cubic-bezier(0.81,0.41,0.13,0.74) 0.5s infinite alternate;
  animation: kSacDd 0.6s cubic-bezier(0.81,0.41,0.13,0.74) 0.5s infinite alternate;
  -webkit-transform-origin: center right;
  -ms-transform-origin: center right;
  transform-origin: center right;
}

.c6 .os-gray {
  -webkit-animation: fXlVAV 0.6s cubic-bezier(0.81,0.41,0.13,0.74) 0.5s infinite alternate;
  animation: fXlVAV 0.6s cubic-bezier(0.81,0.41,0.13,0.74) 0.5s infinite alternate;
  -webkit-transform-origin: center right;
  -ms-transform-origin: center right;
  transform-origin: center right;
}

.c6 .os-yellow {
  -webkit-animation: hbytGi 0.6s cubic-bezier(0.81,0.41,0.13,0.74) 0.6s infinite alternate;
  animation: hbytGi 0.6s cubic-bezier(0.81,0.41,0.13,0.74) 0.6s infinite alternate;
  -webkit-transform-origin: center right;
  -ms-transform-origin: center right;
  transform-origin: center right;
}

.c6 .os-blue {
  -webkit-animation: jfsUaa 0.6s cubic-bezier(0.81,0.41,0.13,0.74) 0.5s infinite alternate;
  animation: jfsUaa 0.6s cubic-bezier(0.81,0.41,0.13,0.74) 0.5s infinite alternate;
  -webkit-transform-origin: center;
  -ms-transform-origin: center;
  transform-origin: center;
}

.c5 {
  display: inline-block;
  vertical-align: -.125em;
  overflow: hidden;
}

.c4 {
  height: 2.5rem;
  width: 2.5rem;
  padding: 0.4rem;
  color: #818181;
  margin-right: 1.4rem;
}

.c0 {
  top: 12rem;
  margin-top: 0;
  padding: 0;
  position: -webkit-sticky;
  position: sticky;
  width: 43.5rem;
  background-color: #f1f1f1;
  box-shadow: 0.2rem 0 0.2rem 0 rgba(0,0,0,0.1);
  z-index: 2;
  height: calc(100vh - 18rem);
}

.c0 > .c7 {
  margin: 0;
  padding: 0;
}

.c0 > .c7::before {
  display: none;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
}

.c1 {
  overflow: hidden;
  right: 0;
  top: 0;
  position: absolute;
  -webkit-transition: opacity 0.2s;
  transition: opacity 0.2s;
  width: 100%;
  height: 100%;
  background-color: #f1f1f1;
  -webkit-transition: opacity 0.5s 0.3s,-webkit-transform 0.2s 0.2s;
  -webkit-transition: opacity 0.5s 0.3s,transform 0.2s 0.2s;
  transition: opacity 0.5s 0.3s,transform 0.2s 0.2s;
}

.c3 {
  border: none;
  padding: 0;
  margin: 0;
  background: transparent;
}

@media screen and (max-width:75em) {
  .c0 {
    width: 100%;
    margin-top: 0;
    top: 15.3rem;
    padding: 0;
  }
}

@media screen and (max-width:75em) {
  .c2 {
    display: none;
  }
}

@media screen and (max-width:75em) {
  .c3 {
    display: none;
  }
}

<div
  aria-label="Search results sidebar"
  className="c0"
  data-testid="search-results-sidebar"
>
  <div
    aria-label="Loading results"
    className="c1"
  >
    <div
      className="c2"
    >
      <button
        className="c3"
        onClick={[Function]}
      >
        <svg
          aria-hidden="true"
          className="c4 c5"
          fill="currentColor"
          focusable="false"
          viewBox="0 0 352 512"
        >
          <path
            d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
            fill="currentColor"
          />
        </svg>
      </button>
    </div>
    <svg
      className="c6"
      id="Layer_1"
      version="1.1"
      viewBox="0 0 57.6 39.1"
      x="0px"
      y="0px"
    >
      <path
        className="os-blue"
        d="M38.4,22c0,0.4-0.3,0.7-0.7,0.7H13.5c-0.4,0-0.7-0.3-0.7-0.7v-1.4c0-0.4,0.3-0.7,0.7-0.7h24.2 c0.4,0,0.7,0.3,0.7,0.7V22z"
        id="os-blue"
        style={
          Object {
            "fill": "#00246a",
          }
        }
      />
      <path
        className="os-yellow"
        d="M41.8,17.8c0,0.4-0.3,0.7-0.7,0.7l-25.4,0.8c-0.4,0-0.7-0.3-0.7-0.7l-0.1-2.1 c0-0.4,0.3-0.7,0.7-0.7L41,15c0.4,0,0.7,0.3,0.7,0.7L41.8,17.8z"
        id="os-yellow"
        style={
          Object {
            "fill": "#f4d019",
          }
        }
      />
      <path
        className="os-gray"
        d="M34.8,14c0,0.4-0.3,0.7-0.7,0.7H11.6c-0.4,0-0.7-0.3-0.7-0.7v-3.5c0-0.4,0.3-0.7,0.7-0.7h22.5 c0.4,0,0.7,0.3,0.7,0.7V14z"
        id="os-gray"
        style={
          Object {
            "fill": "#5e6062",
          }
        }
      />
      <path
        className="os-orange"
        d="M39,8.1c0,0.3-0.4,0.5-1,0.5H8.6c-0.5,0-1-0.2-1-0.5V6.9c0-0.3,0.4-0.5,1-0.5H38 c0.5,0,1,0.2,1,0.5V8.1z"
        id="os-orange"
        style={
          Object {
            "fill": "#f47641",
          }
        }
      />
      <path
        className="os-green"
        d="M43.9,5c0,0.7-0.6,1.2-1.2,1.1L15,4.8c-0.7,0-1.2-0.6-1.1-1.2L14,1.1c0-0.7,0.6-1.2,1.2-1.1 l27.7,1.3c0.7,0,1.2,0.6,1.1,1.2L43.9,5z"
        id="os-green"
        style={
          Object {
            "fill": "#77af42",
          }
        }
      />
      <path
        className="os-type"
        d="M3.1,30.2c-1.8,0-3.1,1.3-3.1,3.1c0,1.8,1.3,3.1,3.1,3.1c1.8,0,3.1-1.3,3.1-3.1 C6.2,31.5,4.9,30.2,3.1,30.2z M3.1,35.7c-1.4,0-2.3-1.1-2.3-2.4c0-1.3,0.9-2.4,2.3-2.4c1.4,0,2.3,1.1,2.3,2.4 C5.4,34.6,4.5,35.7,3.1,35.7z M10.9,30.2c-0.9,0-1.9,0.4-2.4,1.2h0v-1H7.7v8.8h0.8v-3.9h0C9,36,10,36.4,10.9,36.4 c1.8,0,3.1-1.3,3.1-3.1C14,31.5,12.7,30.2,10.9,30.2z M10.9,35.7c-1.4,0-2.5-1.1-2.5-2.4c0-1.3,1.1-2.4,2.5-2.4 c1.4,0,2.3,1.1,2.3,2.4C13.2,34.6,12.3,35.7,10.9,35.7z M18.3,30.2c-1.7,0-2.9,1.3-2.9,3.1c0,1.8,1.2,3.1,3,3.1c1,0,2-0.4,2.6-1.3 l-0.6-0.5c-0.4,0.6-1.2,1-2,1c-1.8,0-2.2-1.6-2.2-2.2h5v-0.5C21.1,31.6,20.1,30.2,18.3,30.2z M16.1,32.8c0-0.3,0.5-1.9,2.1-1.9 c1.1,0,2,0.9,2,1.9H16.1z M27.9,32.7v3.5h-0.8v-3.5c0-1-0.3-1.9-1.5-1.9c-1.1,0-2,0.8-2,2.4v2.9h-0.8V32c0-0.4-0.1-1.4-0.1-1.7h0.8 c0,0.4,0,0.9,0.1,1h0c0.3-0.7,1.1-1.2,2-1.2C27.4,30.2,27.9,31.4,27.9,32.7z M34.2,34.4c0,1.5-1.4,2-2.6,2c-0.9,0-1.8-0.2-2.4-0.9 l1-1c0.4,0.4,0.8,0.8,1.5,0.8c0.4,0,1-0.2,1-0.7c0-1.3-3.3-0.3-3.3-2.6c0-1.4,1.2-2,2.5-2c0.8,0,1.7,0.3,2.2,0.9l-1,0.9 c-0.3-0.4-0.7-0.6-1.2-0.6c-0.4,0-0.9,0.2-0.9,0.7C30.9,32.9,34.2,32,34.2,34.4z M37.5,30.1h1.7v1.3h-1.7v2.7c0,0.6,0.2,1,0.9,1 c0.3,0,0.6-0.1,0.8-0.2v1.3c-0.3,0.2-0.9,0.2-1.2,0.2c-1.6,0-2-0.7-2-2.2v-2.8h-1.3v-1.3H36v-1.8h1.5V30.1z M43,30 c-1,0-1.9,0.3-2.6,1l0.8,0.8c0.4-0.4,1-0.6,1.6-0.6c0.8,0,1.4,0.4,1.4,1.1v0.2h-0.4c-1.5,0-3.9,0.1-3.9,2.1c0,1.2,1.1,1.8,2.2,1.8 c0.8,0,1.5-0.3,2-1h0v0.8h1.4v-3.5C45.6,31.8,45.6,30,43,30z M44.1,33.8c0,0.9-0.5,1.4-1.5,1.4c-0.5,0-1.1-0.2-1.1-0.8 c0-0.9,1.5-0.9,2.3-0.9h0.3V33.8z M50.8,32.9l2.5,3.3h-1.9L49.8,34l-1.6,2.2h-1.8l2.5-3.3l-2.1-2.8h1.9l1.2,1.8l1.3-1.8h1.7 L50.8,32.9z M57.6,30.1v1.7h-0.3v-1.5h0l-0.6,1.5h-0.2l-0.6-1.5h0v1.5h-0.3v-1.7h0.4l0.5,1.3l0.5-1.3H57.6z M53.9,30.1h1.4v0.3h-0.5 v1.5h-0.3v-1.5h-0.5V30.1z"
        id="os-type"
        style={
          Object {
            "fill": "#5e6062",
          }
        }
      />
    </svg>
  </div>
</div>
`;
